queries.sql

```diff
@@ -23,4 +23,68 @@ SELECT * FROM animals
 WHERE name NOT LIKE 'Gabumon';
 
 SELECT * FROM animals
-WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK;
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%___mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+BEGIN; 
+
+DELETE FROM animals; 
+
+ROLLBACK;
+
+-- +++++++++++++++
+
+BEGIN;
+
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT sp1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO sp1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+COMMIT;
+
+-- +++++++++++++++
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals
+WHERE escape_attempts > 0;
+
```

Comment:
- [ ] The query should return how many animals have never tried to escape but you have written a query that returns all animals that have tried to escape. Kindly correct the logic for this query. 👍 😎 

#### Hint: You need to select from the `animals` table where `escape_attempt` is equal to zero. 💻